src/components/ScheduleTable/ScheduleTable.js

```diff
@@ -0,0 +1,198 @@
+import React, { useState, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import { getTimeStringInDoubleFigures } from '../../helpers/getTimeStringInDoubleFigures';
+import TableWithCard from './TableWithCard/TableWithCard';
+import { getDateWithTime } from '../../helpers/getDateWithTime';
+import { addDays } from '../../helpers/addDays';
+
+const ScheduleTable = ({ selectedParams, onClickHandler }) => {
+  const lessons = useSelector((state) => state.lessons.lessons);
+
+  const times = [9, 10, 11, 12, 14, 15, 16, 17];
+  const days = ['mon', 'tue', 'wed', 'thu', 'fri', 'sat', 'sun'];
+
+  let initWeekLessons = {};
+  times.forEach((time, index) => {
+    const startTimeString = getTimeStringInDoubleFigures(time);
+    const endTimeString = getTimeStringInDoubleFigures(time + 1);
+    const lesson = `lesson${index + 1}`;
+    initWeekLessons[lesson] = { slot: `${startTimeString} - ${endTimeString}` };
+    days.forEach((day) => {
+      initWeekLessons[lesson][day] = {
+        startTime: null,
+        endTime: null,
+      };
+    });
+  });
+
+  const [weekLessons, setWeekLessons] = useState(initWeekLessons);
+
+  const columns = [
+    { field: 'slot', headerName: 'Время', width: 100 },
+    { field: 'mon', headerName: 'Пн', width: 50 },
+    { field: 'tue', headerName: 'Вт', width: 50 },
+    { field: 'wed', headerName: 'Ср', width: 50 },
+    { field: 'thu', headerName: 'Чт', width: 50 },
+    { field: 'fri', headerName: 'Пт', width: 50 },
+    { field: 'sat', headerName: 'Сб', width: 50 },
+    { field: 'sun', headerName: 'Вс', width: 50 },
+  ];
+
+  const getLesson = (startTime) => {
+    if (startTime >= 9 && startTime < 10) {
+      return 'lesson1';
+    } else if (startTime >= 10 && startTime < 11) {
+      return 'lesson2';
+    } else if (startTime >= 11 && startTime < 12) {
+      return 'lesson3';
+    } else if (startTime >= 12 && startTime < 13) {
+      return 'lesson4';
+    } else if (startTime >= 14 && startTime < 15) {
+      return 'lesson5';
+    } else if (startTime >= 15 && startTime < 16) {
+      return 'lesson6';
+    } else if (startTime >= 16 && startTime < 17) {
+      return 'lesson7';
+    } else if (startTime >= 17 && startTime < 18) {
+      return 'lesson8';
+    }
+  };
+
+  useEffect(() => {
+    const monday = new Date(selectedParams.startTime);
+    setWeekLessons((prev) => {
+      const copyLessons = { ...prev };
+      times.forEach((time, index) => {
+        const lesson = `lesson${index + 1}`;
+        days.forEach((day, dayIndex) => {
+          copyLessons[lesson][day] = {
+            startTime: getDateWithTime(addDays(monday, dayIndex), time, 0),
+            endTime: getDateWithTime(addDays(monday, dayIndex), time + 1, 0),
+          };
+        });
+      });
+      return copyLessons;
+    });
+  }, [selectedParams]);
```

Comment:
при изменении параметров обнуляется вся таблица с имеющимся расписанием уроков
```suggestion
  }, [selectedParams.startTime, selectedParams.groupId]);
```

Reply:
исправлен